lib/src/from_handlers.dart

```diff
@@ -67,12 +67,9 @@ class _StreamTransformer<S, T> implements StreamTransformer<S, T> {
       controller.onResume = () => subscription?.resume();
     }
     controller.onCancel = () {
-      if (controller.hasListener || subscription == null) {
-        return new Future.value();
-      }
-      var toCancel = subscription;
+      if (controller.hasListener || subscription == null) return;
```

Comment:
The `controller.hasListener` will always be false in the `onCancel` call.
Even a broadcast stream doesn't get an `onCancel` callback for each listener, it only gets one when there is no listener left.